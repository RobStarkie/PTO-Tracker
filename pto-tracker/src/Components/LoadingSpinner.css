.loading-spinner{
    background-color: rgba(83, 83, 83, 0.842);
    border: 2px solid rgba(68, 68, 68, 0.432);
    backdrop-filter: blur(0px);
    opacity: 0%;
    height: 100%;
    width: 100%;
    border-radius: 10px;
    position: absolute;
    z-index: -5;
    top: 0;
    left: 0;
    transition: opacity 0.5s;
  }
  
  .loading-spinner.show {
    opacity: 100%;
    z-index: 5;
    backdrop-filter: blur(10px);
  }
  
  .lds-ring {
    display: inline-block;
    position: absolute;
    width: 80px;
    height: 80px;
    right: 40%;
    bottom: 45%;
  }
  .lds-ring div {
    box-sizing: border-box;
    display: block;
    position: absolute;
    width: 64px;
    height: 64px;
    margin: 8px;
    border: 8px solid #ffffff;
    border-radius: 50%;
    animation: lds-ring 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
    border-color: #ffffff transparent transparent transparent;
  }
  .lds-ring div:nth-child(1) {
    animation-delay: -0.45s;
  }
  .lds-ring div:nth-child(2) {
    animation-delay: -0.3s;
  }
  .lds-ring div:nth-child(3) {
    animation-delay: -0.15s;
  }
  @keyframes lds-ring {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  