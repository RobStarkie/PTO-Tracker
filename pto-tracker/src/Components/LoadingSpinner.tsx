import React from 'react';
import "./LoadingSpinner.css"

interface LoadingSpinnerProps {
    isLoading : boolean
}

const LoadingSpinner: React.FC<LoadingSpinnerProps> = ({ isLoading }) => {

    return (
        <div className={`loading-spinner ${isLoading ? 'show' : ''}`}><div className="lds-ring"><div></div><div></div><div></div><div></div></div></div>
    );
};

export default LoadingSpinner;
