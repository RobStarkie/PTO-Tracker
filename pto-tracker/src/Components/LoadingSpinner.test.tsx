import React from 'react';
import { render } from '@testing-library/react';
import LoadingSpinner from './LoadingSpinner';
import './LoadingSpinner.css'; 

describe('LoadingSpinner', () => {
  it('renders correctly when loading', () => {
    const { asFragment } = render(<LoadingSpinner isLoading={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly when not loading', () => {
    const { asFragment } = render(<LoadingSpinner isLoading={false} />);
    expect(asFragment()).toMatchSnapshot();
  });
});