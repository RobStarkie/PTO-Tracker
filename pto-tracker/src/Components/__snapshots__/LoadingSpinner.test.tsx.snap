// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingSpinner renders correctly when loading 1`] = `
<DocumentFragment>
  <div
    class="loading-spinner show"
  >
    <div
      class="lds-ring"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;

exports[`LoadingSpinner renders correctly when not loading 1`] = `
<DocumentFragment>
  <div
    class="loading-spinner "
  >
    <div
      class="lds-ring"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</DocumentFragment>
`;
